query-engine/driver-adapters/src/proxy.rs

```diff
@@ -581,34 +580,50 @@ impl TransactionProxy {
     pub fn new(js_transaction: &JsObject) -> napi::Result<Self> {
         let commit = js_transaction.get_named_property("commit")?;
         let rollback = js_transaction.get_named_property("rollback")?;
-        let dispose = js_transaction.get_named_property("dispose")?;
         let options = js_transaction.get_named_property("options")?;
 
         Ok(Self {
             commit,
             rollback,
-            dispose,
             options,
+            closed: AtomicBool::new(false),
         })
     }
 
     pub fn options(&self) -> &TransactionOptions {
         &self.options
     }
 
+    /// Commits the transaction via the driver adapter.
+    ///
+    /// Cancelation safety: [`TransactionProxy::closed`] is only set after the call to JS finishes,
```

Comment:
I'm now thinking that it should be the other way around: if there are no other yield points before calling the JS function, the call should always make it to the JS side even if the future is dropped, so there's no need to attempt rolling back (and probably even incorrect for SQLite/libSQL since it uses a single connection).